core/cas-server-core-web-api/src/main/java/org/apereo/cas/web/support/filters/ResponseHeadersEnforcementFilter.java

```diff
@@ -79,12 +79,18 @@ public class ResponseHeadersEnforcementFilter extends AbstractSecurityFilter imp
      * Consent security policy.
      */
     public static final String INIT_PARAM_CONTENT_SECURITY_POLICY = "contentSecurityPolicy";
+    
+    /**
+     * Static resources file extension values.
+     */
+    public static final String INIT_PARAM_CACHE_CONTROL_STATIC_RESOURCES = "cacheControlStaticResources";
 
-    private static final Pattern CACHE_CONTROL_STATIC_RESOURCES_PATTERN = 
-                    Pattern.compile("^.+\\.(css|js|png|txt|jpg|ico|jpeg|bmp|gif)$", Pattern.CASE_INSENSITIVE);
 
     private final Object lock = new Object();
 
+
+    private Pattern cacheControlStaticResourcesPattern = Pattern.compile("^.+\\.(css|js|png|txt|jpg|ico|jpeg|bmp|gif)$", Pattern.CASE_INSENSITIVE);
```

Comment:
This assignment should likely be removed. You should check to see if this value is defined in the config and build a pattern based on that (as you do now). If none is defined, use the default value you have here to build the pattern.

Reply:
If no value is defined in config, it uses default value defined in HttpHeadersRequestProperties.java, which results in the same pattern as on this line. 
I added this line, because in lots of tests, for example verifyCacheControl in [RegisteredServiceResponseHeadersEnforcementFilterTests](https://github.com/Ontik/cas/blob/e494ba7475f9f65b1049d9b01450ad9111daa2f0/core/cas-server-core-web/src/test/java/org/apereo/cas/web/RegisteredServiceResponseHeadersEnforcementFilterTests.java#L48),
the filter function is used without initializing the filter first, which means the pattern is empty, so everything is matched as static resources. Because it is used in lot of tests like this, I found it easier to add a default value that is used when the filter is not initialized, than to add initializing of the filter in all the tests.

Reply:
If that is the case, then please update the test to use the appropriate initialization method. There is no need for dangling pattern that would never in reality be used.

Reply:
Fixed.